editors/code/src/highlighting.ts

```diff
@@ -32,12 +69,25 @@ export class Highlighter {
             tag: string,
             textDecoration?: string
         ): [string, vscode.TextEditorDecorationType] => {
-            const color = new vscode.ThemeColor('ralsp.' + tag);
-            const decor = vscode.window.createTextEditorDecorationType({
-                color,
-                textDecoration
-            });
-            return [tag, decor];
+            const rule = scopesMapper.toRule(tag, scopes.find);
```

Comment:
Nit: this is a bit convoluted (`scopesMapper.toRule(tag, scopes.find)` vs. `scopesMapper.toRule(scopes.find(tag));`.

Reply:
Yeah I feel uneasy about that as well, thanks for the suggestion!

Reply:
@lnicola Just dawned on me they're not really the same thing.
I don't want a tight coupling between mapping and scopes.
The alternative is more verbosity at the call site; 
```typescript
scopesMapper.find(tag)
        .map(scopes.find)
        .filter(rule => rule !== undefined)[0];
```

I think the current approach is nicer now that I think about it.